src/linear_algebra/Hermitian_inner_product.lean

```diff
@@ -0,0 +1,669 @@
+{ norm := abs,
+  dist_eq := by intros; refl, 
+  norm_mul := abs_mul,}
+
+noncomputable instance herm_inner_product_space.normed_space : 
+normed_space ℂ α := 
+{ norm := herm_norm,
+  dist_eq := by intros; refl,
+  norm_smul := herm_norm_smul}
+
+@[simp] lemma herm_norm_zero : 
+|(0 : α)| = 0 := @norm_zero α _
+
+lemma norm_smul_I {W : Type*} [normed_space ℂ W] (x : W) :
+∥I • x∥ = ∥x∥ :=
+begin
+  rw norm_smul,
+  unfold norm,
+  rw [abs_I, field.one_mul], 
+end
+
+@[simp] lemma herm_norm_smul_I (x : α) :
+|I • x| = |x| := norm_smul_I x 
+
+lemma norm_ne_zero_iff_ne_zero {G : Type*} [normed_group G] {g : G} : 
+∥g∥ ≠ 0 ↔ g ≠ 0 := 
+⟨ λ H, (iff_false_left H).mp (norm_eq_zero g), 
+  λ H, (iff_false_right H).mp (norm_eq_zero g)⟩ 
+
+theorem parallelogram_law (x y : α) :
+|x + y|^2 + |x - y|^2 = 2*|x|^2 + 2*|y|^2 :=
+begin
+  dunfold herm_norm,
+  rw [sqr_sqrt (inprod_self_re_nonneg (x + y)), sqr_sqrt (inprod_self_re_nonneg (x - y)),
+    sqr_sqrt (inprod_self_re_nonneg x), sqr_sqrt (inprod_self_re_nonneg y)],
+  suffices : (x ₀ x).re + ((x ₀ x).re + ((y ₀ y).re + (y ₀ y).re)) = 2 * (x ₀ x).re + 2 * (y ₀ y).re,
+  { rw [inprod_add_left, inprod_add_right, inprod_add_right,
+      inprod_sub_left, inprod_sub_right, inprod_sub_right,
+      add_re, add_re, sub_re, sub_re, sub_re],
+    simpa },
+  { rw [←conj_inprod y, conj_re],
+    ring }
+end
+
+lemma inprod_self_add (x y : α) :
+(x + y) ₀ (x + y) = (x ₀ x) + (y ₀ y) + (x ₀ y) + (y ₀ x) :=
+begin
+  rw [inprod_add_left, inprod_add_right, inprod_add_right],
+  simpa [field.add_assoc, field.add_comm],
+end
+
+def is_normalised (x : α) := |x| = 1
+
+noncomputable def normalise (x : α) := (↑|x|⁻¹ : ℂ) • x 
+
+lemma normalise_normalises (x : α) (ho : x ≠ 0) : 
+is_normalised (normalise x) :=
+begin
+  dunfold is_normalised,
+  dunfold normalise,
+  dunfold herm_norm,
+  rw [inprod_smul_left, inprod_smul_right, conj_of_real,
+    mul_re, mul_re, of_real_im, zero_mul, zero_mul,
+    sub_zero, sub_zero, of_real_re, ←sqrt_one,
+    sqrt_inj (mul_nonneg (inv_nonneg.mpr (sqrt_nonneg (x ₀ x).re)) (mul_nonneg (inv_nonneg.mpr (sqrt_nonneg (x ₀ x).re)) (inprod_self_re_nonneg x))) (zero_le_one), 
+    ←sqrt_inv, ←field.mul_assoc],
+  rw mul_self_sqrt,
+  rw field.inv_mul_cancel, 
+    exact (ne_zero_im_zero_imp_re_ne_zero (inprod_self_ne_zero_iff.mpr ho) (inprod_self_im x)),
+    exact (inv_nonneg.mpr (inprod_self_re_nonneg x)),
+end
+
+def normalise_set :
+set α → set α := image(λ x, (↑|x|⁻¹ : ℂ) • x)
+
+lemma normalise_set_normalises 
+(A : set α) (Ho : (0 : α) ∉ A) : 
+∀ x ∈ normalise_set(A), is_normalised x :=
+begin
+  dunfold normalise_set,
+  dunfold image, 
+  intros,
+  have Ha : ∃ (a : α), a ∈ A ∧ normalise a = x,
+  { rw mem_set_of_eq at H, 
+    exact H },
+  cases Ha with a Hl,
+  rw ←Hl.right,
+  have ho : a ≠ 0,
+  { intros h,
+    rw h at Hl,
+    exact Ho Hl.left },
+  exact normalise_normalises a ho,
+end
+
+def herm_ortho (x y : α) : Prop := x ₀ y = 0 
```

Comment:
If you are still in the `herm_inner_product_space` namespace, I wouldn't mind calling this `orthogonal`.